src/middlewared/middlewared/plugins/interface/capabilities_freebsd.py

```diff
@@ -0,0 +1,43 @@
+import netif
+
+from middlewared.service import Service
+
+from .capabilities_base import InterfaceCapabilitiesBase
+
+
+class InterfaceService(Service, InterfaceCapabilitiesBase):
+
+    class Config:
+        namespace_alias = 'interfaces'
+
+    async def nic_capabilities(self):
+        return [c for c in netif.InterfaceCapability.__members__]
+
+    async def to_disable_evil_nic_capabilities(self, check_iface=True):
+        nics = set(await self.middleware.call('jail.nic_capability_checks', None, check_iface))
+        nics.update(await self.middleware.call('vm.device.nic_capability_checks', None, check_iface))
+        return list(nics)
+
+    def enable_capabilities(self, iface, capabilities):
+        enabled = []
+        iface = netif.get_interface(iface)
+        for capability in map(lambda c: getattr(netif.InterfaceCapability, c), capabilities):
+            current = iface.capabilities
+            if capability in current:
+                continue
+            try:
+                iface.capabilities = current | {capability}
+            except OSError:
+                pass
+            else:
+                enabled.append(capability.name)
+        if enabled:
+            self.middleware.logger.debug(f'Enabled {",".join(enabled)} capabilities for {iface}')
+
+    def disable_capabilities(self, iface, capabilities):
+        self.middleware.call_sync('interface._get_instance', iface)
```

Comment:
Let's please use `.get_instance` in favour of `._get_instance` as private methods can cause issues  with service parts